tests/test_virtual_chassis.py

```diff
@@ -24,3 +26,96 @@ def test_connectivity(self, vct):
                 _, out = dv.runcmd(['sh', "-c", "ping -c 5 -W 0 -q %s" % ip])
                 print(out)
                 assert '5 received' in out
+
+    def test_voq_switch(self, vct):
+        """ Test VOQ switch objects configuration """
+        dvss = vct.dvss
+        for name in dvss.keys():
+            if not name.startswith("supervisor"):
+                print("VOQ Switch test for {}".format(name))
+                dvs = dvss[name]
+                #Get the config info
+                config_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+                metatbl = swsscommon.Table(config_db, "DEVICE_METADATA")
+
+                status, cfg_switch_id = metatbl.hget("localhost", "switch_id")
+                assert status, "Got error in getting switch_id from CONFIG_DB DEVICE_METADATA"
+
+                status, cfg_max_cores = metatbl.hget("localhost", "max_cores")
+                assert status, "Got error in getting max_cores from CONFIG_DB DEVICE_METADATA"
+                
+                cfgsptbl = swsscommon.Table(config_db, "SYSTEM_PORT")
+                cfgspkeys = cfgsptbl.getKeys()
+                sp_count = len(cfgspkeys)
+
+                asic_db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+                tbl = swsscommon.Table(asic_db, "ASIC_STATE:SAI_OBJECT_TYPE_SWITCH")
+                keys = list(tbl.getKeys())
+                switch_oid_key = keys[0]
+
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_TYPE")
+                assert status, "Got error while getting switch type"
+                assert value == "SAI_SWITCH_TYPE_VOQ", "Switch type is not VOQ"
+                
+                status, value = tbl.hget(switch_oid_key, "SAI_SWITCH_ATTR_SWITCH_ID")
+                assert status, "Got error while getting switch id"
+                assert value == cfg_switch_id, "VOQ switch id is invalid"
```

Comment:
There is nothing wrong with this code. I just wanted to ask since I couldn't find the related code in this PR or in the repo. Probably it's in another PR.

Where do we specify `switch_type` which is `SAI_SWITCH_TYPE_VOQ` for the test? I assume it should be some where in

tests/virtual_chassis/1/default_config.json
{
    "DEVICE_METADATA": {
        "localhost": {
            "hostname" : "lc1",
            "instance_name": "Linecard1",
            "connect_to_chassis_db" : 1,
            "chassis_db_address" : "10.8.1.200",
            "inband_address" : "10.8.1.1/24",
            "switch_type" : "voq"
        }
    },

And where does the code read this file and parse `switch_type`?

Thanks.

Reply:
No, nevermind. I found the answers.